packages/mobile/src/paymentRequest/IncomingPaymentRequestListItem.tsx

```diff
@@ -1,115 +1,148 @@
-  onPay = () => {
-    const { id, amount, comment: reason, requester: recipient, addressValidationType } = this.props
+  const transactionData: TransactionDataInput = {
+    reason: comment,
+    recipient: requester,
+    amount: new BigNumber(amount),
+    type: TokenTransactionType.PayRequest,
+    firebasePendingRequestUid: id,
+  }
+  const { recipient } = transactionData
+  const { e164PhoneNumber } = recipient
+  const addressValidationType = getAddressValidationType(recipient, secureSendPhoneNumberMapping)
 
-    const transactionData: TransactionDataInput = {
-      reason,
-      recipient,
-      amount: new BigNumber(amount),
-      type: TokenTransactionType.PayRequest,
-      firebasePendingRequestUid: id,
+  const onPay = () => {
+    if (e164PhoneNumber) {
+      setIsLoading(true)
+      // Need to check latest mapping to prevent user from accepting fradulent requests
+      dispatch(fetchAddressesAndValidate(e164PhoneNumber))
+    } else {
+      navigateToNextScreen()
     }
 
     ValoraAnalytics.track(HomeEvents.notification_select, {
       notificationType: NotificationBannerTypes.incoming_tx_request,
       selectedAction: NotificationBannerCTATypes.pay,
     })
-
-    if (addressValidationType && addressValidationType !== AddressValidationType.NONE) {
-      navigate(Screens.ValidateRecipientIntro, { transactionData, addressValidationType })
-    } else {
-      navigate(Screens.SendConfirmation, { transactionData })
-    }
   }
 
-  onPaymentDecline = () => {
-    const { id } = this.props
+  const onPaymentDecline = () => {
     ValoraAnalytics.track(HomeEvents.notification_select, {
       notificationType: NotificationBannerTypes.incoming_tx_request,
       selectedAction: NotificationBannerCTATypes.decline,
     })
-    this.props.declinePaymentRequest(id)
-    Logger.showMessage(this.props.t('requestDeclined'))
+    dispatch(declinePaymentRequest(id))
+    Logger.showMessage(t('requestDeclined'))
   }
 
-  getCTA = () => {
-    return [
-      {
-        text: this.props.t('global:send'),
-        onPress: this.onPay,
-      },
-      {
-        text: this.props.t('global:decline'),
-        onPress: this.onPaymentDecline,
-      },
-    ]
+  const navigateToNextScreen = () => {
+    if (
+      addressValidationType === AddressValidationType.FULL ||
+      addressValidationType === AddressValidationType.PARTIAL
+    ) {
+      navigate(Screens.ValidateRecipientIntro, { transactionData, addressValidationType })
+    } else {
+      navigate(Screens.SendConfirmation, { transactionData })
+    }
   }
 
-  render() {
-    const { requester, id, comment, t } = this.props
-    const name = requester.displayName
-    const amount = {
-      value: this.props.amount,
-      currencyCode: CURRENCIES[CURRENCY_ENUM.DOLLAR].code,
+  React.useEffect(() => {
+    const prevIsFetchingAddresses = prevIsFetchingAddressesRef.current
+    prevIsFetchingAddressesRef.current = isFetchingAddresses
```

Comment:
i might be wrong but will want to put extra attention here due to JS using references not values.